pkg/types/json_binary_functions.go

```diff
@@ -134,6 +135,16 @@ func unquoteJSONString(s string) (string, error) {
 				}
 				char, size, err := decodeEscapedUnicode(hack.Slice(s[i+1 : i+5]))
 				if err != nil {
+					// For `surrogate pair`, it uses two `\\uxxx` to encode a character.
+					if size < 0 && len(s) >= i+10 && s[i+5] == '\\' && s[i+6] == 'u' {
+						char, size, err = decodeEscapedUnicode(append(hack.Slice(s[i+1:i+5]), hack.Slice(s[i+7:i+11])...))
```

Comment:
seems `append` will modify `s`'s underlying array, I'm not sure if it will cause some problems after caller returns.